src/utils.js

```diff
@@ -0,0 +1,55 @@
+
```

Comment:
To avoid obnoxious merge conflicts, can we refactor this after we get these three huge PRs in?

#52 created